neurovault/apps/statmaps/management/commands/benchmark.py

```diff
@@ -0,0 +1,104 @@
+from django.core.management.base import BaseCommand, CommandError
+from neurovault.apps.statmaps.tests.utils import clearDB
+from neurovault.apps.statmaps.models import Comparison, Similarity, User, Collection, Image
+from neurovault.apps.statmaps.utils import get_existing_comparisons
+from neurovault.apps.statmaps.tests.utils import  save_statmap_form
+
+import os
+import gc
+import timeit
+import numpy as np
+
+class Timer:
+    def __init__(self, timer=None, disable_gc=False, verbose=True):
+        if timer is None:
+            timer = timeit.default_timer
+        self.timer = timer
+        self.disable_gc = disable_gc
+        self.verbose = verbose
+        self.start = self.end = self.interval = None
+    def __enter__(self):
+        if self.disable_gc:
+            self.gc_state = gc.isenabled()
+            gc.disable()
+        self.start = self.timer()
+        return self
+    def __exit__(self, *args):
+        self.end = self.timer()
+        if self.disable_gc and self.gc_state:
+            gc.enable()
+        self.interval = self.end - self.start
+        if self.verbose:
+            print('time taken: %f seconds' % self.interval)
+
+class Command(BaseCommand):
+    args = '<times_to_run>'
+    help = 'bench'
+
+    def handle(self, *args, **options):
+        # TODO: check if /bench folder exists. If not, donwload the full dataset.
```

Comment:
You can also consider uploading a tarball with benchmark maps on figshare
On Jun 9, 2016 12:29 PM, "Asier Erramuzpe" <notifications@github.com> wrote:

> In neurovault/apps/statmaps/management/commands/benchmark.py
> <https://github.com/NeuroVault/NeuroVault/pull/517#discussion_r66417216>:
>
> > +        self.start = self.timer()
> > +        return self
> > +    def __exit__(self, *args):
> > +        self.end = self.timer()
> > +        if self.disable_gc and self.gc_state:
> > +            gc.enable()
> > +        self.interval = self.end - self.start
> > +        if self.verbose:
> > +            print('time taken: %f seconds' % self.interval)
> > +
> > +class Command(BaseCommand):
> > +    args = '<times_to_run>'
> > +    help = 'bench'
> > +
> > +    def handle(self, *args, **options):
> > +        # TODO: check if /bench folder exists. If not, donwload the full dataset.
>
> Ok, I will take a look into it and see how to manage.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NeuroVault/NeuroVault/pull/517/files/6465933bf16cff0d45dd02d67876a3cc8e4efedf#r66417216>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAOkp-8-ttHbnyYxDqw8bFhhN1Rf6l49ks5qJ-r3gaJpZM4IqvB2>
> .
>